File: easybluetoothble/src/main/java/com/srise/easybluetoothble/impl/ReceiveMessageBusiness.java
package com.srise.easybluetoothble.impl;


import android.util.Log;

import com.srise.easybluetoothble.Interface.IReceiveDataEngine;
import com.srise.easybluetoothble.Interface.IReceiveMessage;
import com.srise.easybluetoothble.bean.BaseMessage;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;


public class ReceiveMessageBusiness implements IReceiveMessage {
    private static final String TAG = "ReceiveMessageBusiness";
    private IReceiveDataEngine mReceiveDataEngine;
    private ExecutorService mExecutorService;

    public ReceiveMessageBusiness() {
        mReceiveDataEngine = new BleReceiveDataEngine();
        mExecutorService = Executors.newFixedThreadPool(1);
    }


    @Override
    public void receiveMessage(BaseMessage message) {
        Log.d(TAG, " receiveMessage");
        mExecutorService.submit(new ReceiveTask(message));
    }

    public void destroy() {
        Log.d(TAG, " destroy");
        mExecutorService.shutdownNow();
    }

    @Override
    public void resetData() {
        Log.d(TAG, " resetData");
        mReceiveDataEngine.resetData();
    }

    private class ReceiveTask implements Runnable {
        private BaseMessage mMessage;

        public ReceiveTask(BaseMessage message) {
            mMessage = message;
        }

        @Override
        public void run() {
            Log.d(TAG, " ReceiveTask run");

            mReceiveDataEngine.recreateDataFrame(mMessage);
        }
    }
}

File: ktlionble/src/main/java/c/example/dell/ktlionble/impl/ReceiveMessageBusiness.kt
package c.example.dell.ktlionble.impl

import android.util.Log
import c.example.dell.ktlionble.bean.BaseMessage
import c.example.dell.ktlionble.inter.IReceiveDataEngine
import c.example.dell.ktlionble.inter.IReceiveMessage
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

class ReceiveMessageBusiness : IReceiveMessage {
    companion object {
        const val TAG = "ReceiveMessageBusiness"
    }

    private var mReceiveDataEngine: IReceiveDataEngine = BleReceiveDataEngine()
    private var mExecutorService: ExecutorService = Executors.newFixedThreadPool(1)

    override fun destroy() {
        Log.d(TAG, " destroy")
        mExecutorService.shutdownNow()
    }

    override fun resetData() {
        Log.d(TAG, " resetData")
        mReceiveDataEngine.resetData()
    }

    override fun receiveMessage(message: BaseMessage) {
        Log.d(TAG, " receiveMessage")
        mExecutorService.submit(ReceiveTask(message))
    }

    inner class ReceiveTask() : Runnable { //TODO
        private lateinit var message: BaseMessage

        constructor(message: BaseMessage) : this() {
            this.message = message
        }

        override fun run() {
            Log.d(TAG, " ReceiveTask run");

            mReceiveDataEngine.recreateDataFrame(message);
        }
    }
}